cfme/tests/storage/test_volume_snapshot.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+
+from cfme import test_requirements
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.utils.log import logger
+from cfme.utils.wait import wait_for
+
+
+pytestmark = [
+    test_requirements.storage,
+    pytest.mark.ignore_stream("upstream"),
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.provider([OpenStackProvider], scope='module')
+]
+
+STORAGE_SIZE = 1
+
+
+@pytest.yield_fixture(scope='module')
```

Comment:
**REQUIRED**: This should be function scoped as there are tests that delete it.

Reply:
@psav: now  `volume` fixture scope  `module` is right as it's for complete test not going to delete in any test. and I created new fixture l.e. `snapshot` having scope `function`. As per @mfalesni and IRC discussion fixture should be used for `setup` and `teardown` so I am not including `snapshot` fixture in `crud test`.